# Match Case Statements

To implement switch-case like characteristics very similar to if-else functionality, we use a match case in python. 

**A match statement will compare a given variable's value to different shapes, also referred to as the pattern. The main idea is to keep on comparing the variable with all the present patterns until it fits into one.**

The match case consists of three entities:
1. The match keyword 
2. One or more case clauses
3. Expression for each case

The case clause consists of a pattern to be matched to the variable, a condition to be evaluated if the pattern matches, and a set of statement to be executed if the pattern matches. 


In [1]:
# match variable_name: 
#    case 'pattern1' : //statement1
#    case 'pattern2' : //statement2
#   ....
#   case 'pattern n' : //statement n"""

In [1]:
x = 4
# x is the variable to match
match x: 
    #if x is 0
    case 0:
        print("x is zero")
    # case with if-condition
    case 4 if x % 2 == 0:
        print("x % 2 == 0 and cse is 4")
    # Empty case with if-condition
    case _ if x <10: 
        print("x is < 10")
    #default case (will only be matched if the above cases were not matched)
    # so it is basically just an else:
    case _:
        print(x)
        

x % 2 == 0 and cse is 4


In [2]:
x = 4
match x:
    case 0: 
        print("x is zero")
    case 4 if x % 2 == 0:
        print("x % 2 == 0 and case is 4")
    case _ if x < 10:
        print("x is < 10")
    

x % 2 == 0 and case is 4


In [3]:
from platform import python_version
print(python_version())

3.12.2


updating python notebook from 3.9 to 3.12

# ERROR solution

The `match` case function is available from Python 3.10 onwards. Therefore, in Python 3.9, it will result in an error.

To utilize the `match` case function, I installed the latest version of Python, 3.12.

Here are the steps to install the `match` case function in an Anaconda environment:

First, create a new environment because installing it in the base environment might lead to conflicts with other installed packages.

1. `mamba create -n py3.12_jupyter python=3.12`

Secondly, install Python 3.12 in the `py3.12_jupyter` environment:

2. `mamba install -n py3.12_jupyter python=3.12 jupyter`

Finally, activate the environment to use Jupyter Notebook:

3. `conda activate py3.12_jupyter`

Now, you can use the `match` function in this environment.

(Note: Each time you want to use Python 3.12, you need to activate this environment.)
